cyipopt/scipy_interface.py

```diff
@@ -533,3 +538,39 @@ def minimize_ipopt(fun,
                           nfev=problem.nfev,
                           njev=problem.njev,
                           nit=problem.nit)
+
+def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
+                       bounds, constraints, tol, callback, options):
+    # basic input validation for minimize_ipopt
+    if fun is not None and not callable(fun):
+        raise ValueError('`fun` must be callable.')
+    x0 = np.asarray(x0)[()]
+    if not np.issubdtype(x0.dtype, np.number):
+        raise ValueError('`x0` must be a numeric array.')
+    if not np.iterable(args):
+        args = (args,)
```

Comment:
We've discussed how to handle this situation in SciPy (https://github.com/scipy/scipy/issues/15994), but we haven't really reached a conclusion. This implements one of the suggestions given there. Another possibility is to raise an error (instead of wrapping `args` in a tuple) if `args` is not iterable or to be even more strict and require that `args` is a tuple. LMK which you prefer.

Reply:
Well, it wraps in a tuple if it's not already one:
https://github.com/scipy/scipy/blob/c1ed5ece8ffbf05356a22a8106affcd11bd3aee0/scipy/optimize/_minimize.py#L527

Ok, I'll do that.

Reply:
```suggestion
    if not isinstance(args, tuple):
        args = (args,)
```